modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeNetworkClusterFactory.java

```diff
@@ -0,0 +1,94 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.network.scalecube;
+
+import io.scalecube.cluster.Cluster;
+import io.scalecube.cluster.ClusterImpl;
+import io.scalecube.net.Address;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.ignite.network.MessageHandlerHolder;
+import org.apache.ignite.network.NetworkCluster;
+import org.apache.ignite.network.NetworkClusterContext;
+import org.apache.ignite.network.NetworkClusterFactory;
+
+/**
+ * Factory for ScaleCubeNetworkCluster.
+ */
+public class ScaleCubeNetworkClusterFactory implements NetworkClusterFactory {
+    /** Unique name of network member. */
+    private final String localMemberName;
+
+    /** Local port. */
+    private final int localPort;
+
+    /** Network addresses to find another members in cluster. */
+    private final List<String> addresses;
+
+    /**
+     * Member resolver which allows convert {@link org.apache.ignite.network.NetworkMember} to inner ScaleCube type
+     * and otherwise.
+     */
+    private final ScaleCubeMemberResolver memberResolver;
+
+    /**
+     * @param localMemberName Unique name of network member.
+     * @param port Local port.
+     * @param addresses Network addresses to find another members in cluster.
+     */
+    public ScaleCubeNetworkClusterFactory(
+        String localMemberName,
+        int port,
+        List<String> addresses,
+        ScaleCubeMemberResolver memberResolver
+    ) {
+        this.localMemberName = localMemberName;
+        localPort = port;
+        this.addresses = addresses;
+        this.memberResolver = memberResolver;
+    }
+
+    /**
+     * Implementation of {@link NetworkCluster} based on ScaleCube.
+     *
+     * @param memberResolver Member resolve which allows convert {@link org.apache.ignite.network.NetworkMember} to
+     * inner ScaleCube type and otherwise.
+     * @param messageHandlerHolder Holder of all cluster message handlers.
+     * @return {@link NetworkCluster} instance.
+     */
+    @Override public NetworkCluster startCluster(NetworkClusterContext clusterContext) {
+        final MessageHandlerHolder handlerHolder = clusterContext.messageHandlerHolder();
+        Cluster cluster = new ClusterImpl()
+            .handler(cl -> {
+                return new ScaleCubeMessageHandler(cl, memberResolver, handlerHolder);
+            })
+            .config(opts -> opts
+                .memberAlias(localMemberName)
+                .transport(trans -> {
+                    return trans.port(localPort)
+                        .connectTimeout(1000_000_000)
+                        .messageCodec(new ScaleCubeMessageCodec(clusterContext.messageMappers()));
+                })
+            )
+            .membership(opts -> opts.seedMembers(addresses.stream().map(Address::from).collect(Collectors.toList())))
```

Comment:
I think this address conversion should happen in the constructor so that we can be informed about malformed addresses earlier